src/sprightly.ts

```diff
@@ -129,4 +129,17 @@ function sprightlyAsync(
   })
 }
 
+sprightly.__express = (
+  path: string,
+  data: Data,
+  callback: Function,
+  options?: Options,
```

Comment:
Passing an `options` argument here isn't very useful because the idea is that express will be calling this when a template file is referenced by requiring its extension and accessing the `.__express` property. So assuming our files extension is `.spy`, express, internally, will do something similar to the following when the template file `osid-sucks.spy` is referenced:
```js
const render = require("spy").__express;

render(
  path,
  options, // options have the data passed by the user and express options in the same object
  callback
);
```
So express won't pass any user-defined options to sprightly.


As you probably have deduced from everything I said above, defining a `.__express` property is a bad requirement from my side since I don't want to define a specific extension for sprightly's templates and use regular `.html` files instead. This means users have to define their template engine of choice manually like this:
```js
import sprightly from "sprightly/express";

app.engine('html', sprightly({
  // engine options
}))
```
So what you have to do now is to create a default-exported API at `sprightly/express` or `sprightly/adapters/express` that is a factory function which when invoked returns the engine function loaded with the options passed by the user and accepts the arguments passed by express mentioned above, since express will be calling this returned engine function.
Please inspect the `cache` property passed inside express's `options` and see whether to use that, the user-defined `cache` option, or maybe some combination of both?